pxtcompiler/emitter/service.ts

```diff
@@ -701,8 +701,14 @@ namespace ts.pxtc.service {
                 if (this.opts.fileSystem[fn] != v) {
                     this.fileVersions[fn] = (this.fileVersions[fn] || 0) + 1
                 }
-            })
-            this.opts = o
+            });
+            // shallow copy, but deep copy the file system
```

Comment:
Was hitting a bug with this where it was manipulating the file system service host's file systems without updating file versions (we don't hit this in the webapp as the opts are always stringified / destringified when being passed to the worker). Can move this to be cloned in the cli but little cost to doing a shallow clone here